base_layer/wallet/src/output_manager_service/recovery/standard_outputs_recoverer.rs

```diff
@@ -0,0 +1,153 @@
+// Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+// following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following
+// disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+// following disclaimer in the documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote
+// products derived from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+// INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+// USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+use crate::output_manager_service::{
+    error::OutputManagerError,
+    storage::{
+        database::{OutputManagerBackend, OutputManagerDatabase},
+        models::DbUnblindedOutput,
+    },
+    MasterKeyManager,
+};
+use blake2::Digest;
+use log::*;
+use std::sync::Arc;
+use tari_core::transactions::{
+    transaction::{TransactionOutput, UnblindedOutput},
+    types::{CryptoFactories, PrivateKey, PublicKey},
+};
+use tari_crypto::{
+    common::Blake256,
+    inputs,
+    keys::PublicKey as PublicKeyTrait,
+    script,
+    tari_utilities::{hex::Hex, ByteArray},
+};
+
+const LOG_TARGET: &str = "wallet::output_manager_service::recovery";
+
+pub(crate) struct StandardUtxoRecoverer<TBackend: OutputManagerBackend + 'static> {
+    master_key_manager: Arc<MasterKeyManager<TBackend>>,
+    factories: CryptoFactories,
+    db: OutputManagerDatabase<TBackend>,
+}
+
+impl<TBackend> StandardUtxoRecoverer<TBackend>
+where TBackend: OutputManagerBackend + 'static
+{
+    pub fn new(
+        master_key_manager: Arc<MasterKeyManager<TBackend>>,
+        factories: CryptoFactories,
+        db: OutputManagerDatabase<TBackend>,
+    ) -> Self {
+        Self {
+            master_key_manager,
+            factories,
+            db,
+        }
+    }
+
+    /// Attempt to rewind all of the given transaction outputs into unblinded outputs. If they can be rewound then add
+    /// them to the database and increment the key manager index
+    pub async fn scan_and_recover_outputs(
```

Comment:
```suggestion
    pub async fn scan_recover_and_add_outputs(
```

This function now scans, recovers, and adds the outputs it finds. 
I am not completely for this as this function now does more than one thing, but we should at least update the name to reflect that.

Reply:
I will argue that the verb `recover` is the process of adding it to the database. The method does to two things, `scans` which is going through and trying to rewind an output and `recover` which is adding it to the db and updating the keymanager. I thought about splitting the second part out but decided that I couldn't think of a use case for every allowing these to happen in isolation and that the interface should only allow them to be done together.

Reply:
I respectfully disagree, "recover" here already means "recover for spending" which means add them to your wallet.